app/controllers/controllers.js

```diff
@@ -22,11 +22,26 @@ exports.getAllEndpoints = (req, res, next) => {
     })
 } 
 
-exports.getAllArticles = (req, res, next) => { 
-    return selectAllArticles()
-    .then((result) => { 
-        res.status(200).send({ articles: result})
-    })
+exports.getAllArticles = (req, res, next) => {
+    if(Object.keys(req.query).length) {
+        const topic = req.query.topic
```

Comment:
Could this logic for changing your db query dependent on presence of topic be moved into the model/simplified somehow to make your code more DRY? Would be better practice to handle it there